cloud/google/configure_service.sh

```diff
@@ -0,0 +1,97 @@
+#!/bin/bash
+set -e
+DIR=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
+GCLOUD="gcloud"
+if [ "$1" == "" ]; then
+  echo "Usage: $0 <config-file>"
+  exit 1
+fi
+CONFIG_FILE="$1"
+
+. ${DIR}/config.sh ${CONFIG_FILE}
+
+case "${INSTANCE_TYPE}" in
+  "mirror")
+    eval $(typeset -A -p MIRROR_ZONES | sed 's/MIRROR_ZONES=/NODE_ZONES=/')
+    eval $(typeset -A -p MIRROR_MACHINES | \
+           sed 's/MIRROR_MACHINES=/NODE_MACHINES=/')
+    export NODE_NUM_REPLICAS=${MIRROR_NUM_REPLICAS}
+    export NODE_ZONES
+    ;;
+  "log")
+    eval $(typeset -A -p LOG_ZONES | sed 's/LOG_ZONES=/NODE_ZONES=/')
+    eval $(typeset -A -p LOG_MACHINES | \
+           sed 's/LOG_MACHINES=/NODE_MACHINES=/')
+    export NODE_NUM_REPLICAS=${LOG_NUM_REPLICAS}
+    export NODE_ZONES
+   ;;
+  *)
+    echo "Unknown INSTANCE_TYPE: ${INSTANCE_TYPE}"
+    exit 1
+esac
+
+ZONE_LIST=$(echo ${NODE_ZONES[*]} | tr " " "\n" | sort | uniq )
+
+echo "============================================================="
+echo "Creating network rules..."
+gcloud compute http-health-checks create get-sth-check \
+    --port 80 \
+    --request-path /ct/v1/get-sth
+gcloud compute firewall-rules create ${INSTANCE_TYPE}-node-80 \
+    --allow tcp:80 \
+    --target-tags ${INSTANCE_TYPE}-node
+
+for zone in ${ZONE_LIST}; do
+  gcloud preview instance-groups \
+      --zone ${zone} create "${INSTANCE_TYPE}-group-${zone}" \
+      --network "default" &
+done
+wait
+
+for i in `seq 0 $((${NODE_NUM_REPLICAS} - 1))`; do
+  gcloud preview instance-groups \
+      --zone ${NODE_ZONES[${i}]} instances \
+      --group "${INSTANCE_TYPE}-group-${NODE_ZONES[${i}]}" \
+      add ${NODE_MACHINES[${i}]} &
+done
+wait
+
+gcloud compute addresses create "${INSTANCE_TYPE}-ip" \
+    --global
+export EXTERNAL_IP=$(gcloud compute addresses list "${INSTANCE_TYPE}-ip" |
+                     awk -- "/${INSTANCE_TYPE}-ip/ {print \$2}")
+echo "Service IP: ${EXTERNAL_IP}"
+
+
+gcloud compute backend-services create "${INSTANCE_TYPE}-lb-backend" \
+    --http-health-check "get-sth-check" \
+    --timeout "30"
+
+for zone in ${ZONE_LIST}; do
+  gcloud compute backend-services add-backend "${INSTANCE_TYPE}-lb-backend" \
+    --group "${INSTANCE_TYPE}-group-${zone}" \
+    --zone ${zone} \
+    --balancing-mode "UTILIZATION" \
+    --capacity-scaler "1" \
```

Comment:
Those are just the defaults that it gives you behind-the-scenes when you configure this via the cloud console.  There's doco for this and all other commands behind `--help` flags and on-line.